api/devices/schema.py

```diff
@@ -15,12 +19,18 @@ class Meta:
 class CreateDevice(graphene.Mutation):
     class Arguments:
         name = graphene.String(required=True)
+        room_id = graphene.Int(required=True)
         device_type = graphene.String(required=True)
         location = graphene.String()
-        resource_id = graphene.Int(required=True)
     device = graphene.Field(Devices)
 
+    @Auth.user_roles('Admin')
     def mutate(self, info, **kwargs):
+        location_query = location_join_room()
+        room_location = location_query.filter(RoomModel.id == kwargs['room_id']).first()  # noqa: E501
```

Comment:
Looks good. I just have a suggestion: why not use `room_location = location_join_room().filter(...).first()` directly instead of doing this first `location_query = location_join_room()`.

Reply:
Thanks, this is actually a cleaner way. Let me try it.